### PR TITLE
converts emag actions to BALLOON ALERTS (+ a tiny bit of machine tool interaction)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

all emag messages give feedback to the player via balloon_alerts instead of to_chats.

minor stuff I did in machinery + hailer, if you don't like it, I can remove it.

## Why It's Good For The Game

I'm testing the water for what is acceptable with balloon alerts because I really like them but i'm not 100% sure a good use case is. I think emag feedback being made concise to what exactly the emag action is in a small balloon feedback is really nice and keeps chat less cluttered.

## Changelog
:cl:
qol: changed emag actions putting a message in your chat to balloon alerts.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
### come on now
image

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -647,15 +647,16 @@
...
 		I.play_tool_sound(src, 50)
 		var/prev_anchored = anchored
 		//as long as we're the same anchored state and we're either on a floor or are anchored, toggle our anchored state
 		if(I.use_tool(src, user, time, extra_checks = CALLBACK(src, .proc/unfasten_wrench_check, prev_anchored, user)))
 			if(!anchored && ground.is_blocked_turf(exclude_mobs = TRUE, source_atom = src))
+				balloon_alert(user, "interrupted!")
 				to_chat(user, "<span class='notice'>You fail to secure [src].</span>")
 				return CANT_UNFASTEN
-			to_chat(user, "<span class='notice'>You [anchored ? "un" : ""]secure [src].</span>")
+			balloon_alert(user, "[anchored ? "un" : ""]secured [src].")
```
Comment (Mothblocks) [inline code/game/machinery/_machinery.dm:659]:
```suggestion
			balloon_alert(user, "[anchored ? "un" : ""]secured [src]")
```

File: code/game/machinery/fat_sucker.dm [hunk 2]
```diff
@@ -215,5 +215,5 @@
 		return
 	start_at = 100
 	stop_at = 0
-	to_chat(user, "<span class='notice'>You remove the access restrictions and lower the automatic ejection threshold!</span>")
+	balloon_alert(user, "restrictions lifted, automatic ejection threshold lowered")
```
Comment (Mothblocks) [inline code/game/machinery/fat_sucker.dm:218]: Not necessary, but this honestly might look best as two balloon alerts, "automatic ejection threshold lowered" delayed a bit after the other.

File: code/game/machinery/firealarm.dm [hunk 3]
```diff
@@ -120,8 +120,8 @@
 	obj_flags |= EMAGGED
 	update_appearance()
 	if(user)
-		user.visible_message("<span class='warning'>Sparks fly out of [src]!</span>",
-							"<span class='notice'>You emag [src], disabling its thermal sensors.</span>")
+		balloon_alert(user, "thermal sensors disabled")
+	visible_message("<span class='warning'>Sparks fly out of [src]!</span>")
```
Comment (Mothblocks) [inline code/game/machinery/firealarm.dm:124]: Can't we just ACTUALLY have sparks?

Comment (Mothblocks) [reply code/game/machinery/firealarm.dm:124]: I guess this is emag so it should be stealthy...meh.

File: code/modules/clothing/glasses/hud.dm [hunk 4]
```diff
@@ -38,7 +38,7 @@
 	if(obj_flags & EMAGGED)
 		return
 	obj_flags |= EMAGGED
-	to_chat(user, "<span class='warning'>PZZTTPFFFT</span>")
```
Comment (Mothblocks) [inline code/modules/clothing/glasses/hud.dm:41]: that sound u make when u shit

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -534,7 +534,7 @@
 	. = ..()
 	if(can_flashlight && attached_light) //if it has a light but can_flashlight is false, the light is permanently attached.
 		I.play_tool_sound(src)
-		to_chat(user, "<span class='notice'>You unscrew [attached_light] from [src].</span>")
+		balloon_alert(user, "[attached_light] removed from [src]")
```
Comment (Mothblocks) [inline code/modules/clothing/head/helmet.dm:537]:
```suggestion
		balloon_alert(user, "removed [attached_light]")
```

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -560,7 +560,7 @@
 		return
 	attached_light.on = !attached_light.on
 	attached_light.update_brightness()
-	to_chat(user, "<span class='notice'>You toggle the helmet light [attached_light.on ? "on":"off"].</span>")
+	balloon_alert(user, "light toggled [attached_light.on ? "on":"off"]")
```
Comment (Mothblocks) [inline code/modules/clothing/head/helmet.dm:563]: Do we really need this? A big light comes from your head.

File: code/modules/mining/abandoned_crates.dm [hunk 7]
```diff
@@ -103,6 +103,7 @@
 
 /obj/structure/closet/crate/secure/loot/emag_act(mob/user)
 	if(locked)
+		balloon_alert(user, "anti-tamper system activated") //idiot
```
Comment (Mothblocks) [inline code/modules/mining/abandoned_crates.dm:106]: Do we need this? The feedback is provided in the uh...y'know...explosion.

Comment (Mothblocks) [reply code/modules/mining/abandoned_crates.dm:106]: Guess it doesn't matter too much, but eh.

File: code/modules/mob/living/silicon/robot/robot_defense.dm [hunk 8]
```diff
@@ -319,20 +319,20 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 		return
 	if(!opened)//Cover is closed
 		if(locked)
-			to_chat(user, "<span class='notice'>You emag the cover lock.</span>")
+			balloon_alert(user, "cover lock unlocked")
```
Comment (Mothblocks) [inline code/modules/mob/living/silicon/robot/robot_defense.dm:322]:
```suggestion
			balloon_alert(user, "unlocked cover")
```

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm [hunk 9]
```diff
@@ -35,6 +35,7 @@
 /mob/living/simple_animal/bot/secbot/ed209/emag_act(mob/user)
 	..()
 	icon_state = "ed209[on]"
+	balloon_alert(user, "weapon free mode set")
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/ed209bot.dm:38]:
```suggestion
	balloon_alert(user, "weapon-free mode set")
```

File: code/modules/mob/living/simple_animal/bot/firebot.dm [hunk 10]
```diff
@@ -128,7 +128,7 @@
 	..()
 	if(emagged == 2)
 		if(user)
-			to_chat(user, "<span class='danger'>[src] buzzes and beeps.</span>")
+			balloon_alert(user, "bot set to \"fight fire with fire\" mode")
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/firebot.dm:131]: cute

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 11]
```diff
@@ -145,6 +145,7 @@
 	..()
 	if(emagged == 2)
 		if(user)
+			balloon_alert(user, "opposite day uploaded")
 			to_chat(user, "<span class='danger'>[src] buzzes and beeps.</span>")
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:149]: Don't need this anymore.

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:149]: In the future we should replace this with a low radius, chatless Runechat emote.

File: code/modules/power/singularity/emitter.dm [hunk 12]
```diff
@@ -388,7 +388,8 @@
 	locked = FALSE
 	obj_flags |= EMAGGED
 	if(user)
-		user.visible_message("<span class='warning'>[user.name] emags [src].</span>", "<span class='notice'>You short out the lock.</span>")
+		balloon_alert(user, "lock shorted out")
+		user.visible_message("<span class='warning'>[user.name] emags [src].</span>")
```
Comment (Mothblocks) [inline code/modules/power/singularity/emitter.dm:392]: lmao is this really what they went with

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I'm very happy with this.

File: code/game/machinery/firealarm.dm (hunk 3 above)
Comment (Qustinnus) [reply code/game/machinery/firealarm.dm:124]: You can just have silent sparks

File: code/modules/mining/abandoned_crates.dm (hunk 7 above)
Comment (tralezab) [reply code/modules/mining/abandoned_crates.dm:106]: my thought process was that all these emag interactions have a balloon alert so it was there for consistency

and yeah it's harmless

File: code/modules/power/singularity/emitter.dm (hunk 12 above)
Comment (tralezab) [reply code/modules/power/singularity/emitter.dm:392]: yeah...

File: code/game/machinery/fat_sucker.dm (hunk 2 above)
Comment (tralezab) [reply code/game/machinery/fat_sucker.dm:218]: added

File: code/modules/clothing/head/helmet.dm (hunk 6 above)
Comment (tralezab) [reply code/modules/clothing/head/helmet.dm:563]: you know what, flashlights don't use feedback, so neither will this.

File: code/modules/mob/living/simple_animal/bot/firebot.dm (hunk 10 above)
Comment (tralezab) [reply code/modules/mob/living/simple_animal/bot/firebot.dm:131]: ty ty

File: code/game/machinery/fat_sucker.dm [hunk 13]
```diff
@@ -215,5 +215,10 @@
 		return
 	start_at = 100
 	stop_at = 0
-	to_chat(user, "<span class='notice'>You remove the access restrictions and lower the automatic ejection threshold!</span>")
+	balloon_alert(user, "restrictions lifted")
+	addtimer(CALLBACK(src, .proc/finish_emag_act, user), 0.5 SECONDS)
```
Comment (Mothblocks) [inline code/game/machinery/fat_sucker.dm:219]: Remove `finish_emag_act`, and do:

```suggestion
	addtimer(CALLBACK(src, .proc/balloon_alert, user, "automatic ejection threshold lowered"), 0.5 SECONDS)
```

If this doesn't work, try `/atom/proc/balloon_alert`.

**Comment by maxymax13** [comment]
what does this look like

File: code/game/machinery/firealarm.dm (hunk 3 above)
Comment (cacogen) [reply code/game/machinery/firealarm.dm:124]: Sparks create light and set things on fire but yeah it should have them or remove the message entirely

**Comment by Mothblocks** [comment]
https://github.com/tgstation/tgstation/pull/59130

File: code/modules/holodeck/computer.dm [hunk 14]
```diff
@@ -373,11 +373,12 @@ and clear when youre done! if you dont i will use :newspaper2: on you
 	if(obj_flags & EMAGGED)
 		return
 	if(!LAZYLEN(emag_programs))
+		balloon_alert(user, "no programs to corrupt")
 		to_chat(user, "[src] does not seem to have a card swipe port. It must be an inferior model.")
```
Comment (maxymax13) [inline code/modules/holodeck/computer.dm:377]: why is this still here

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm [hunk 15]
```diff
@@ -110,15 +110,15 @@
...
+/datum/species/ethereal/proc/on_emag_act(mob/living/carbon/human/emagged, mob/user)
 	if(emageffect)
 		return
 	emageffect = TRUE
 	if(user)
-		to_chat(user, "<span class='notice'>You tap [H] on the back with your card.</span>")
-	H.visible_message("<span class='danger'>[H] starts flickering in an array of colors!</span>")
-	handle_emag(H)
-	addtimer(CALLBACK(src, .proc/stop_emag, H), 2 MINUTES) //Disco mode for 2 minutes! This doesn't affect the ethereal at all besides either annoying some players, or making someone look badass.
+		user.balloon_alert(user, "disco time for [emagged]")
```
Comment (maxymax13) [inline code/modules/mob/living/carbon/human/species_types/ethereal.dm:118]: 😆

File: code/modules/modular_computers/computers/item/tablet.dm [hunk 16]
```diff
@@ -50,9 +50,10 @@
 
 /obj/item/modular_computer/tablet/nukeops/emag_act(mob/user)
 	if(!enabled)
+		balloon_alert(user, "turn it on first!")
 		to_chat(user, "<span class='warning'>You'd need to turn the [src] on first.</span>")
```
Comment (maxymax13) [inline code/modules/modular_computers/computers/item/tablet.dm:54]: why keep

Comment (tralezab) [reply code/modules/modular_computers/computers/item/tablet.dm:54]: true my bad

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm (hunk 15 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/species_types/ethereal.dm:118]: ty ty

File: code/game/machinery/firealarm.dm (hunk 3 above)
Comment (zxaber) [reply code/game/machinery/firealarm.dm:124]: Sparks to light the plasma in the room you're disabling fire alarms. Sounds almost like an intended combo.

File: code/game/machinery/fat_sucker.dm [hunk 17]
```diff
@@ -216,5 +216,5 @@
 	start_at = 100
 	stop_at = 0
 	balloon_alert(user, "restrictions lifted")
-	addtimer(CALLBACK(src, .proc/balloon_alert, user, "automatic ejection threshold lowered"), 0.5 SECONDS)
+	addtimer(CALLBACK(src, /atom.proc/balloon_alert, user, "automatic ejection threshold lowered"), 0.5 SECONDS)
```
Comment (LemonInTheDark) [inline code/game/machinery/fat_sucker.dm:219]: Why are you doing this?

**Comment by Dorsisdwarf** [comment]
I personally think only overt actions should have onscreen popups to avoid confusion about what other players can and cant see you do but im overruled :(

**Comment by Mothblocks** [comment]
It's *already* confusing. `to_chat` and `visible_message` have no obvious distinction to the player on what can be seen by you and what can be seen by everyone.

**Comment by cacogen** [comment]
I feel like visible_message or its equivalent didn't used to have "You do X" but rather "Y does X" which made it more obvious when other people were seeing it.

**Comment by Mothblocks** [comment]
+1'ing Lemon's comment.

**Comment by Sparkezel** [comment]
Whats the range of that bubble?

**Comment by Mothblocks** [comment]
`balloon_alert` is only to you. `balloon_alert_to_viewers` is everyone in view range.

**Comment by cacogen** [comment]
the cat comes for us all

**Comment by Mothblocks** [comment]
@tralezab 👀

**Comment by tralezab** [comment]
oh, uh, yeah my bad. i'll finish this

**Pull request closed without merging**